### PR TITLE
Split pkg/restic package

### LINKED ISSUE #5055: Split the restic package
In the code of the package `.../pkg/restic` there are some generic code that should be used also by restic for example `backupper.go` `repository_manager.go` ....

We should consider split such files into another package, say `.../pkg/podvolumes` which then can reference `.../pkg/restic` and the new `.../pkg/kopia` to provide support for different repository backends.

### PR DESCRIPTION
**Pull request by ywk253100**
This commit splits the pkg/restic package into several packages to support Kopia integration works

Fixes #5055

Signed-off-by: Wenkai Yin(尹文开) <yinw@vmware.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/restic/common.go [hunk 1]
```diff
@@ -52,99 +52,15 @@ const (
...
-	// pod volume backups when they're for a PVC.
-	PVCNameAnnotation = "velero.io/pvc-name"
-
 	// VolumesToBackupAnnotation is the annotation on a pod whose mounted volumes
 	// need to be backed up using restic.
 	VolumesToBackupAnnotation = "backup.velero.io/backup-volumes"
 
 	// VolumesToExcludeAnnotation is the annotation on a pod whose mounted volumes
 	// should be excluded from restic backup.
 	VolumesToExcludeAnnotation = "backup.velero.io/backup-volumes-excludes"
```
Comment (Lyndon-Li) [inline pkg/restic/common.go:61]: Do we need to move these two ones to uploader as well?

File: pkg/restic/mocks/repository_manager.go [hunk 2]
```diff
@@ -73,15 +74,15 @@ func (_m *RepositoryManager) InitRepo(repo *v1.BackupRepository) error {
 }
 
 // NewBackupper provides a mock function with given fields: _a0, _a1
-func (_m *RepositoryManager) NewBackupper(_a0 context.Context, _a1 *v1.Backup) (restic.Backupper, error) {
+func (_m *RepositoryManager) NewBackupper(_a0 context.Context, _a1 *v1.Backup) (uploader.Backupper, error) {
```
Comment (Lyndon-Li) [inline pkg/restic/mocks/repository_manager.go:77]: I see we've moved the `NewBackupper` and `NewRestorer` out of repository_manager, so may also need to move the mocks?

File: pkg/uploader/util.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	}
+	return volume.Projected != nil || volume.DownwardAPI != nil
+}
+
+// getPodSnapshotAnnotations returns a map, of volume name -> snapshot id,
+// of all restic snapshots for this pod.
+// TODO(2.0) to remove
+// Deprecated: we will stop using pod annotations to record restic snapshot IDs after they're taken,
+// therefore we won't need to check if these annotations exist.
+func getPodSnapshotAnnotations(obj metav1.Object) map[string]string {
```
Comment (Lyndon-Li) [inline pkg/uploader/util.go:95]: Change the comments here? Replace "restic"

File: pkg/restic/mocks/repository_manager.go (hunk 2 above)
Comment (ywk253100) [reply pkg/restic/mocks/repository_manager.go:77]: It's the mock for `RepositoryManager`, although we moved the `NewBackerupper` and `NewRestorer`, the `RepositoryManager` still needs to implement these two functions

File: pkg/uploader/util.go (hunk 3 above)
Comment (ywk253100) [reply pkg/uploader/util.go:103]: It's hard to replace all "restic" in the code/comments until we reach an agreement on what should be changed to. Here  is already a open [issue](https://github.com/vmware-tanzu/velero/issues/5094) to track the task, let's do the replacement when handling the issue

Comment (Lyndon-Li) [reply pkg/uploader/util.go:103]: Ok, thanks.

**Review by Lyndon-Li — APPROVED** [review]

**Review by Lyndon-Li — APPROVED** [review]

**Comment by reasonerjt** [comment]
Sorry for the delayed response, but how about we rename `pkg/uploader` to `pkg/podvolume` ?

**Review by reasonerjt — APPROVED** [review]
lgtm

**Review by Lyndon-Li — APPROVED** [review]

**Pull request merged**